### PR TITLE
Add Additional Role Claims Functionality

### PR DESCRIPTION
**Pull request by AksharX**
**Add the following functionality to Admin API**
1. Delete all Roles Claims for all Roles that have a specific Claim Type (Input: Role Claim Type)
2. Add a specific Claim to Roles (Input: List<Roles>, ClaimType, ClaimValue)
3. Get all Roles that have a specific Role Claim Type (Input: Role Claim)

I believe this is a good introduction to having more functionality around Role Claims.

### CODE REVIEW

**Comment by skoruba** [comment]
Thanks for your PR, I will check it. 👍

**Comment by ruzanowski** [comment]
That's good input, however, we might add UI to that 👍 
@AksharX possible to resolve conflicts and add simple management to make it fully functional?

File: src/Skoruba.IdentityServer4.Admin.BusinessLogic.Identity/Services/IdentityService.cs [hunk 1]
```diff
@@ -473,5 +473,44 @@ public virtual async Task<IdentityResult> DeleteRoleAsync(TRoleDto role)
...
+        {
+            foreach (var roleId in roleIds)
+            {
+                var roleExists = await IdentityRepository.ExistsRoleAsync(roleId);
+                if (!roleExists) throw new UserFriendlyErrorPageException(string.Format(IdentityServiceResources.RoleDoesNotExist().Description, roleId ), IdentityServiceResources.RoleDoesNotExist().Description);
+            }
+
+            var added = await IdentityRepository.AddClaimToRolesAsync(roleIds, roleClaimType, roleClaimValue);
+
+            await AuditEventLogger.LogEventAsync(new RoleClaimDeletedFromRolesEvent(roleClaimType, roleIds));
```
Comment (ruzanowski) [inline src/Skoruba.IdentityServer4.Admin.BusinessLogic.Identity/Services/IdentityService.cs:502]: Wrong event. Should be: RoleClaimAddedToRolesEvent

File: src/Skoruba.IdentityServer4.Admin.BusinessLogic.Identity/Services/Interfaces/IIdentityService.cs [hunk 2]
```diff
@@ -85,12 +85,18 @@ public interface IIdentityService<TUserDto, TUserDtoKey, TRoleDto, TRoleDtoKey,
...
-
+		Task<TRoleClaimsDto> GetUserRoleClaimsAsync(string userId, string claimSearchText, int page = 1, int pageSize = 10);
+
 		Task<TRoleClaimsDto> GetRoleClaimAsync(string roleId, int claimId);
 
         Task<int> DeleteRoleClaimsAsync(TRoleClaimsDto role);
 
         Task<IdentityResult> DeleteRoleAsync(TRoleDto role);
+
+        Task<int> DeleteClaimForRolesAsync(List<string> roleIds, string roleClaimType);
```
Comment (ruzanowski) [inline src/Skoruba.IdentityServer4.Admin.BusinessLogic.Identity/Services/Interfaces/IIdentityService.cs:96]: IList

File: src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs [hunk 3]
```diff
@@ -360,6 +360,53 @@ public virtual async Task<IdentityResult> DeleteUserAsync(string userId)
             return await UserManager.DeleteAsync(userIdentity);
         }
 
+        public virtual async Task<int> DeleteClaimForRolesAsync(List<string> roleIds, string roleClaimType)
+        {
+            var rolesQ = from r in DbContext.Set<TRole>()
+                join rc in DbContext.Set<TRoleClaim>() on r.Id equals rc.RoleId
+                where rc.ClaimType.Equals(roleClaimType) && roleIds.Contains(r.Id.ToString())
```
Comment (ruzanowski) [inline src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs:367]: roleIds.Contains(r.Id.ToString()) in EFCore3.1 is not translated.

try: roleIds.Contains(((string)(object)r.Id))

**Review by ruzanowski — CHANGES_REQUESTED** [review]

**Pull request closed without merging**